src/libraries.ts

```diff
@@ -72,14 +74,20 @@ export function getImplicitStylesheets(imports: Iterable<string>): Set<string> {
  * library used by FileAttachment) we manually enumerate the needed additional
  * downloads here. TODO Support versioned imports, too, such as "npm:leaflet@1".
  */
-export function getImplicitDownloads(imports: Iterable<string>): Set<string> {
+export function getImplicitDownloads(imports: Iterable<string>, duckdb?: DuckDBConfig): Set<string> {
   const set = setof(imports);
   const implicits = new Set<string>();
   if (set.has("npm:@observablehq/duckdb")) {
     implicits.add("npm:@duckdb/duckdb-wasm/dist/duckdb-mvp.wasm");
     implicits.add("npm:@duckdb/duckdb-wasm/dist/duckdb-browser-mvp.worker.js");
     implicits.add("npm:@duckdb/duckdb-wasm/dist/duckdb-eh.wasm");
     implicits.add("npm:@duckdb/duckdb-wasm/dist/duckdb-browser-eh.worker.js");
+    if (!duckdb) throw new Error("Implementation error: missing duckdb configuration");
+    for (const [name, {source}] of Object.entries(duckdb.extensions)) {
+      for (const platform of duckdb.bundles) {
+        implicits.add(`duckdb:${platform},${name},${source}`);
```

Comment:
Any reason not to follow the same convention that DuckDB does here for custom repositories?

Reply:
I guess one answer is that `name` and `platform` should never contain a comma, making it easier to split — but there’s no guarantee that `source` doesn’t contain a comma, so it’s not safe to `path.split(",")` and expect to get all the parts back out again. I’m going to tweak this a bit to match the DuckDB convention and make it more robust.